ceno_zkvm/src/scheme/prover.rs

```diff
@@ -240,6 +240,7 @@ impl<
 
         let mut points = Vec::new();
         let mut evaluations = Vec::new();
+        let mut trace_idx = 0usize; // tracks the index of circuits with witness columns (num_witin > 0) in PCS data
```

Comment:
We can hide `trace_idx` and make extract_witness_mles return an iterator. It's ok we do it next